software/image_processor.py

```diff
@@ -78,23 +78,29 @@ def start(self):
     def stop(self):
         self.camera.close()
 
-    def analyze_balls(self, t_balls, depth, fragments) -> list:
+    def analyze_balls(self, t_balls, depth, fragments, basket) -> list:
+        kernel = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (5, 5))
```

Comment:
Declaring the kernel on every loop costs some performance, especially since it does not change over time.

Reply:
Moved to constructor.